### ISSUE TITLE
Support annotating variable declaration with Allow.PlainSQL

### ISSUE THREAD

**Issue by victornoel** [jOOQ/jOOQ#9114]
### Use case:

When allowing to use plain SQL, it is usually better to do so at the most restricted scope.
It seems that `Allow.PlainSQL` smallest scope is `METHOD`, which sometimes is not small enough.

It would be better if we could tackle this at the object level.

### Possible solution you'd like to see:

The solution would be to be able to annotate at the declaration level with `LOCAL_VARIABLE`.

Actually I wonder if it wouldn't be needed also on `FIELD` since one can use jooq to initialise the value of a field.

### Possible workarounds:

Create dedicated methods for each desired scope.

### Versions:

- jOOQ: 3.12.0
- Java: 11
- Database (include vendor): N/A
- OS:
- JDBC Driver (include name if inofficial driver):

**Comment by lukaseder** [comment]
Thank you very much for your feedback. Great to see feedback for one of the most underrated jOOQ features :-)

I agree, if it is possible to pull this off with the checker framework and/or ErrorProne, we could allow for this annotation to be placed on smaller scopes.

- `LOCAL_VARIABLE` would mean that the initialisation and re-assignment of such variables would allow for using plain SQL
- `FIELD` could be interesting as well, both for the initialisation and re-assignment of a field

Obviously, we would have to allow `MODULE` as well in the Java 11+ distribution

**Comment by tbroyer** [comment]
Not sure about reassignment.
`@Allow.PlainSQL` is similar to me to `@SuppressWarnings` (you could actually use that with a specific value, instead of a specific annotation) which only suppresses warnings on the field/variable initialization.

**Comment by lukaseder** [comment]
> Not sure about reassignment.

Yes, indeed. It would not work for reassignments, out of the box. Maybe, we could hack our way around the annotation processors to detect reassignments, or check if a variable is (effectively) final and make that a requirement. Or just fail, in case a reassignment does use plain SQL.

**Comment by tbroyer** [comment]
My point was, as a user, just do the same as `@SuppressWarnings`, don't even try to track reassignments.

```java
// passes the check
@Allow.PlainSQL Condition cond = otherCond.and("whatever");

// fails the check
cond = otherCond.and("whatever");

// workaround to pass the check
@Allow.PlainSQL Condition temp = otherCond.and("whatever");
cond = temp;
```

(either the temporary local variable so you have a place to apply the annotation, or annotate the enclosing method)
